sgkit/stats/aggregation.py

```diff
@@ -546,3 +546,129 @@ def sample_stats(
         ]
     )
     return conditional_merge_datasets(ds, variables.validate(new_ds), merge)
+
+
+def infer_call_ploidy(
+    ds: Dataset,
+    *,
+    call_genotype_non_allele: Hashable = variables.call_genotype_non_allele,
+    merge: bool = True,
+) -> Dataset:
+    """Infer the ploidy of each genotype call.
+
```

Comment:
It would be nice to give a little bit more detail here (one sentence) as to how it actually does this